File: caixa/src/main/java/br/com/majority/caixa/modelo/Produto.java
package br.com.majority.caixa.modelo;

import java.math.BigDecimal;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToMany;

@Entity
public class Produto {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;
	
	private String nome;
	
	private String marca;
	
	private String categoria;
	
	private BigDecimal preco;
	
	private BigDecimal quantidade;

	public Integer getId() {
		return id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public String getMarca() {
		return marca;
	}

	public void setMarca(String marca) {
		this.marca = marca;
	}

	public String getCategoria() {
		return categoria;
	}

	public void setCategoria(String categoria) {
		this.categoria = categoria;
	}

	public BigDecimal getPreco() {
		return preco;
	}

	public void setPreco(BigDecimal preco) {
		this.preco = preco;
	}
	
	public void setQuantidade(BigDecimal quantidade) {
		this.quantidade = quantidade;
	}
	
	public BigDecimal getQuantidade() {
		return quantidade;
	}
	
}

File: caixa/src/main/resources/static/js/models/Produto.js
class Produto{
	
	constructor(id, nome, marca, categoria, preco, quantidade){
		
		this._id = id;
		this._nome = nome;
		this._marca = marca;
		this._categoria = categoria;
		this._preco = preco;
		this._quantidade = quantidade;
	}
	
	set id (id){
		
		this._id = id;
	}
	get id (){
		
		return this._id;
	}
	
	set nome (nome){
		
		this._nome = nome;
	}
	get nome (){
		
		return this._nome;
	}
	
	set marca (marca){
		
		this._marca = marca;
	}
	get marca (){
		
		return this._marca;
	}
	
	set categoria (categoria){
		
		this._categoria = categoria;
	}
	get categoria (){
		
		return this._categoria;
	}
	
	set preco (preco){
		
		this._preco = preco;
	}
	get preco (){
		
		return this._preco;
	}
	
	set quantidade (quantidade){
		
		this._quantidade = quantidade;
	}
	get quantidade (){
		
		return this._quantidade;
	}
}